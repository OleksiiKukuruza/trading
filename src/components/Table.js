import PropTypes from 'prop-types';
import styled from 'styled-components';

const Table = styled.div`
  border: 1px solid ${props => props.theme.primary.main};
`;

const Row = styled.div`
  display: flex;

  :not(:last-child) {
    border-bottom: 1px solid ${props => props.theme.background};
  }
`;

const HeaderRow = styled(Row)`
  background-color: ${props => props.theme.background};
`;

const Cell = styled.div`
  flex: ${props => props.size || 1};
  display: flex;
  justify-content: ${props => props.justifyContent || 'flex-start'};
`;

Cell.propTypes = {
  size: PropTypes.number,
  justifyContent: PropTypes.string
};

export { Table, Row, HeaderRow, Cell };
