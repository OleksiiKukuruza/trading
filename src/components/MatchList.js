import React from 'react';
import styled from 'styled-components';
import { Cell, HeaderRow, Row } from './Table';

const StyledList = styled.div`
  margin: ${props => 2 * props.theme.spacingUnit}px;
`;

const StyledListContent = styled.div`
  border: 1px solid ${props => props.theme.background};
`;

const MatchList = ({ matches }) => (
  <StyledList>
    Matches
    <StyledListContent>
      <HeaderRow>
        <Cell>Match Time</Cell>
        <Cell>Price</Cell>
        <Cell>Quantity</Cell>
      </HeaderRow>
      {matches.map(match => (
        <Row key={`${match.sell.id}-${match.buy.id}`}>
          <Cell>{match.time}</Cell>
          <Cell>{(match.buy.price + match.sell.price) / 2}</Cell>
          <Cell>{Math.min(match.sell.quantity, match.buy.quantity)}</Cell>
        </Row>
      ))}
    </StyledListContent>
  </StyledList>
);

export default MatchList;
