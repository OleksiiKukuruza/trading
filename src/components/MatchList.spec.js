import React from 'react';
import { shallow } from 'enzyme';
import MatchList from './MatchList';
import { Row } from './Table';

const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />);

  return {
    wrapper
  };
};

describe('MatchList', () => {
  it('renders without crashing', () => {
    const { wrapper } = setup(MatchList, {
      openModal: () => null,
      closeModal: () => null,
      openedMatch: {
        time: 123,
        sell: { id: 3, price: 50, quantity: 3 },
        buy: { id: 4, price: 60, quantity: 6 }
      },
      matches: [
        {
          time: 123,
          sell: { id: 3, price: 50, quantity: 3 },
          buy: { id: 4, price: 60, quantity: 6 }
        }
      ]
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('handles click on a row', () => {
    const openModal = jest.fn();
    const openedMatch = {
      time: 123,
      sell: { id: 3, price: 50, quantity: 3 },
      buy: { id: 4, price: 60, quantity: 6 }
    };
    const { wrapper } = setup(MatchList, {
      openModal,
      closeModal: () => null,
      openedMatch,
      matches: [openedMatch]
    });
    const firstRow = wrapper.find(Row).at(0);
    firstRow.simulate('click');
    expect(openModal).toHaveBeenCalledWith(openedMatch);
  });
});
