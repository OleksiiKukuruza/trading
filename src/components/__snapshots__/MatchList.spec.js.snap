// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchList renders without crashing 1`] = `
<div>
  Matches
  <styled.div>
    <Styled(styled.div)>
      <styled.div>
        Match Time
      </styled.div>
      <styled.div>
        Price
      </styled.div>
      <styled.div>
        Quantity
      </styled.div>
    </Styled(styled.div)>
    <styled.div
      key="3-4"
      onClick={[Function]}
    >
      <styled.div>
        3:00:00 AM
      </styled.div>
      <styled.div>
        55
      </styled.div>
      <styled.div>
        3
      </styled.div>
    </styled.div>
  </styled.div>
  <MatchModal
    match={
      Object {
        "buy": Object {
          "id": 4,
          "price": 60,
          "quantity": 6,
        },
        "sell": Object {
          "id": 3,
          "price": 50,
          "quantity": 3,
        },
        "time": 123,
      }
    }
    onClose={[Function]}
  />
</div>
`;
